File: app/serializers/supplier.py
from app.models import Supplier


def serialize_supplier(supplier: Supplier) -> dict:
    return {
        "id": supplier.id,
        "tenant_id": supplier.tenant_id,
        "supplier_code": supplier.supplier_code,
        "name": supplier.name,
        "legal_name": supplier.legal_name,
        "contact_person": supplier.contact_person,
        "email": supplier.email,
        "phone": supplier.phone,
        "alternate_phone": supplier.alternate_phone,
        "address_line_1": supplier.address_line_1,
        "address_line_2": supplier.address_line_2,
        "city": supplier.city,
        "county_or_region": supplier.county_or_region,
        "country": supplier.country,
        "postal_code": supplier.postal_code,
        "tax_number": supplier.tax_number,
        "registration_number": supplier.registration_number,
        "payment_terms_days": supplier.payment_terms_days,
        "credit_limit": (
            str(supplier.credit_limit)
            if supplier.credit_limit is not None
            else "0.00"
        ),
        "currency": supplier.currency,
        "notes": supplier.notes,
        "is_active": supplier.is_active,
        "created_at": supplier.created_at.isoformat() if supplier.created_at else None,
        "updated_at": supplier.updated_at.isoformat() if supplier.updated_at else None,
    }

File: frontend/src/types/entities/supplier.ts
/**
 * ============================================================================
 * Hela360 Supplier Entity
 * ============================================================================
 *
 * Canonical frontend representation of the verified backend Supplier
 * serializer payload.
 *
 * The supplier service currently returns backend JSON directly, so this entity
 * uses the backend snake_case field names truthfully until a verified service
 * mapping is introduced.
 *
 * ============================================================================
 */

export interface Supplier {
  id: string;

  tenant_id: string;

  supplier_code: string;

  name: string;

  legal_name: string | null;

  contact_person: string | null;

  email: string | null;

  phone: string | null;

  alternate_phone: string | null;

  address_line_1: string | null;

  address_line_2: string | null;

  city: string | null;

  county_or_region: string | null;

  country: string | null;

  postal_code: string | null;

  tax_number: string | null;

  registration_number: string | null;

  payment_terms_days: number;

  credit_limit: string;

  currency: string;

  notes: string | null;

  is_active: boolean;

  created_at: string | null;

  updated_at: string | null;
}
